addons/bus/static/src/js/crosstabbus.js

```diff
@@ -0,0 +1,350 @@
+     *
+     * @override
+     */
+    updateOption: function () {
+        this._super.apply(this, arguments);
+        this._callLocalStorage('setItem', 'options', this.options);
+    },
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+    /**
+     * Call local_storage service
+     *
+     * @private
+     * @param {string} method (getItem, setItem, removeItem, on)
+     * @param {string} key
+     * @param {any} param
+     * @returns service information
+     */
+    _callLocalStorage: function (method, key, param) {
+        return this.call('local_storage', method, this._generateKey(key), param);
+    },
+    /**
+     * Generates localStorage keys prefixed by bus. (localStorageKey = the name
+     * of this addon), and the sanitized origin, to prevent keys from
+     * conflicting when several bus instances (polling different origins)
+     * co-exist.
+     *
+     * @private
+     * @param {string} key
+     * @returns key prefixed with the origin
+     */
+    _generateKey: function (key, param) {
+        return this.localStorageKey + '.' + this.sanitizedOrigin + '.' + key;
+    },
+    /**
+     * @override
+     */
+    _getLastPresence: function () {
+        return this._callLocalStorage('getItem', 'lastPresence') || this._super();
+    },
+    /**
+     * Check all the time (according to the constants) if the tab is the master tab and
+     * check if it is active. Use the local storage for this checks.
+     *
+     * @see _startElection method
+     */
+    _heartbeat: function () {
+        var self = this;
+        var current = new Date().getTime();
+        var heartbeatValue = this._callLocalStorage('getItem', 'heartbeat') || 0;
+        var peers = this._callLocalStorage('getItem', 'peers', {});
+
+        if ((parseInt(heartbeatValue) + this.heartbeatOutOfDatePeriod) < current) {
+            // Heartbeat is out of date. Electing new master
+            this._startElection();
+            heartbeatValue = this._callLocalStorage('getItem', 'heartbeat') || 0;
+        }
+
+        if (this._isMasterTab) {
+            //walk through all peers and kill old
+            var cleanedPeers = {};
+            for (var peerName in peers) {
+                if (peers[peerName] + this.heartbeatKillOldPeriod > current) {
+                    cleanedPeers[peerName] = peers[peerName];
+                }
+            }
+
+            if (parseInt(heartbeatValue) !== this.lastHeartbeat) {
+                // someone else is also master...
+                // it should not happen, except in some race condition situation.
+                this._isMasterTab = false;
+                this.lastHeartbeat = 0;
+                peers[this.id] = current;
+                this._callLocalStorage('setItem', 'peers', peers);
+                this.stopPolling();
+                this.trigger('no_longer_master');
```

Comment:
What is the purpose of this event? I do not see any `.on('no_longer_master', ...)`